include/field.hxx

```diff
@@ -83,10 +87,10 @@ class Field {
 #endif
 
   virtual Mesh * getMesh() const{
-    if (fieldmesh){
-      return fieldmesh;
+    if (fieldmesh == nullptr) {
```

Comment:
I think we generally prefer the true branch to be the most common case, which I'd guess is that `fieldmesh` is not `nullptr`. Rewriting as `return fieldmesh != nullptr? fieldmesh : bout::globals::mesh` would be one way to go.